### PR TITLE
Add support for Hilt AssistedInject

### PR DESCRIPTION
**Pull request by ghostbear**
The latest Dagger/Hilt version added support for `AssistedInject` with `HiltViewModel`. 

> Add support for using @AssistedInject with @HiltViewModel. (https://github.com/google/dagger/commit/83271776751d7aa4fe8683984682b0016b08bf5e).
>
>https://github.com/google/dagger/releases/tag/dagger-2.49

And this PR adds that functionality to Voyager.

### CODE REVIEW

File: voyager-hilt/src/main/java/cafe/adriel/voyager/hilt/ViewModel.kt
Comment (DevNatan) [inline voyager-hilt/src/main/java/cafe/adriel/voyager/hilt/ViewModel.kt:68]: Please avoid using abbreviation for type parameters with more than one letter (with the exception of VM = ViewModel) thats not clear about what it is
```suggestion
public inline fun <reified VM : ViewModel, F> Screen.getViewModel(
    viewModelProviderFactory: ViewModelProvider.Factory? = null,
    noinline viewModelFactory: (F) -> VM,
```
```suggestion
public inline fun <reified VM : ViewModel, Factory> Screen.getViewModel(
    viewModelProviderFactory: ViewModelProvider.Factory? = null,
    noinline viewModelFactory: (Factory) -> VM,
```

**Review by DevNatan — APPROVED** [review]

**Comment by DevNatan** [comment]
LGTM just wait for @DevSrSouza and @adrielcafe approval 👍🏽

**Comment by DevSrSouza** [comment]
Thanks for the contribution and sample code!

Can you annotate the function with `@Experimental` for now? we should soon work to try to merge all Hilt functions into one, but for now we can provide this as experimental (ABI only)

**Pull request merged**
